contracts/minter-suite/Minters/MinterPolyptych/MinterPolyptychV0.sol

```diff
@@ -0,0 +1,809 @@
+            // as they are known to be implicitly checked by calling `checkDelegateForToken`.
+            bool isValidVault = delegationRegistryContract
+                .checkDelegateForToken(
+                    msg.sender, // delegate
+                    _vault, // vault
+                    genArt721CoreAddress, // contract
+                    _ownedNFTTokenId // tokenId
+                );
+            require(isValidVault, "Invalid delegate-vault pairing");
+            vault = _vault;
+        }
+
+        // we need the new token ID in advance of the randomizer setting a token hash
+        (uint256 _invocations, , , , , ) = genArtCoreContract.projectStateData(
+            _projectId
+        );
+        uint256 _newTokenId = (_projectId * ONE_MILLION) + _invocations;
+
+        address _artist = genArtCoreContract.projectIdToArtistAddress(
+            _projectId
+        );
+
+        // EFFECTS
+
+        // we need to store the new token ID before it is minted so the randomizer can query it
+        IGenArt721CoreContractV3WithRandomizer genArtCoreContractWithRandomizer = IGenArt721CoreContractV3WithRandomizer(
+                _ownedNFTAddress
+            );
+        bytes12 targetHashSeed = genArtCoreContractWithRandomizer
+            .tokenIdToHashSeed(_ownedNFTTokenId);
+        require(targetHashSeed != 0);
+
+        genArtCoreContractWithRandomizer
+            .randomizerContract()
+            .setPolyptychHashSeed(_newTokenId, targetHashSeed);
+
+        // once mint() is called, the polyptych randomizer will either:
+        // 1) assign a random token hash
+        // 2) if configured, obtain the token hash from the `polyptychSeedHashes` mapping
+        tokenId = minterFilter.mint(_to, _projectId, vault);
+
+        // NOTE: delegate-vault handling **ends here**.
+
+        // redundant check against reentrancy
+        bytes12 _assignedHashSeed = genArtCoreContractWithRandomizer
+            .tokenIdToHashSeed(tokenId);
+        require(_assignedHashSeed == targetHashSeed);
+
+        if (_invocations == _projectConfig.maxInvocations) {
+            _projectConfig.maxHasBeenInvoked = true;
+        }
+
+        // INTERACTIONS
+        // require sender to own NFT used to redeem
+        /**
+         * @dev Considered an interaction because calling ownerOf on an NFT
+         * contract. Plan is to only register AB/PBAB NFTs on the minter, but
+         * in case other NFTs are registered, better to check here. Also,
+         * function is non-reentrant, so being extra cautious.
+         */
+        if (msg.sender == _artist) {
+            require(
+                IERC721(_ownedNFTAddress).ownerOf(_ownedNFTTokenId) == vault,
+                "Only owner of NFT"
+            );
+        } else {
+            require(
+                IERC721(_ownedNFTAddress).ownerOf(_ownedNFTTokenId) == _to,
+                "Only owner of NFT"
```

Comment:
I think `vault` and `_to` should be flipped here, but please triple check me on this and let's confirm in tests 😅 😅

Reply:
++